examples/ffva/src/usb/adaptive_rate_adjust.c

```diff
@@ -23,21 +23,27 @@
 #include <rtos_interrupt.h>
 
 #include "platform/app_pll_ctrl.h"
+#include "tusb.h"
 
 #ifndef USB_ADAPTIVE_TASK_PRIORITY
 #define USB_ADAPTIVE_TASK_PRIORITY (configMAX_PRIORITIES-1)
 #endif /* USB_ADAPTIVE_TASK_PRIORITY */
 
-#define DATA_EVENT_QUEUE_SIZE 1
+#define DATA_EVENT_QUEUE_SIZE 2 // For OUT and IN directions
 
 typedef struct usb_audio_rate_packet_desc {
     uint32_t cur_time;
     uint32_t ep_num;
     uint32_t ep_dir;
     size_t xfer_len;
+    bool calc_rate;
 } usb_audio_rate_packet_desc_t;
 
+extern volatile bool mic_interface_open;
```

Comment:
externed from where?

Reply:
I'll add a comment. The set interface callback in usb_audio.c is the only place I can get the interface open information reliably from so doing an extern from there.